jsf-ri/src/main/java/com/sun/faces/component/search/SearchKeywordResolverImplChild.java

```diff
@@ -0,0 +1,113 @@
+ * recipient has the option to distribute your version of this file under
+ * either the CDDL, the GPL Version 2 or to extend the choice of license to
+ * its licensees as provided above.  However, if you add GPL Version 2 code
+ * and therefore, elected the GPL Version 2 license, then the option applies
+ * only if the new code is made subject to such option by the copyright
+ * holder.
+ */
+package com.sun.faces.component.search;
+
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import javax.faces.FacesException;
+import javax.faces.component.UIComponent;
+import javax.faces.component.search.UntargetableComponent;
+import javax.faces.component.search.SearchExpressionContext;
+import javax.faces.component.search.SearchKeywordContext;
+import javax.faces.component.search.SearchKeywordResolver;
+
+public class SearchKeywordResolverImplChild extends SearchKeywordResolver {
+
+    private static final Pattern PATTERN = Pattern.compile("child\\((\\d+)\\)");
+    
+    @Override
+    public void resolve(SearchKeywordContext searchKeywordContext, UIComponent previous, String command) {
+        try {
+            Matcher matcher = PATTERN.matcher(command);
+
+            if (matcher.matches()) {
+
+                int childNumber = Integer.parseInt(matcher.group(1));
+                if (childNumber + 1 > previous.getChildCount()) {
+                    throw new FacesException("Component with clientId \""
+                            + previous.getClientId(searchKeywordContext.getSearchExpressionContext().getFacesContext()) 
+                            + "\" has fewer children as \"" + 
+                              childNumber + "\". Expression: \"" + command + "\"");
+                }
+
+                List<UIComponent> list = previous.getChildren();
+                int count = 0;
+                for (int i = 0; i < previous.getChildCount(); i++) {
+                    if (! (list.get(i) instanceof UntargetableComponent)) {
+                        count++;
+                    }
+                    if (count == childNumber + 1) {
+                        searchKeywordContext.invokeContextCallback(previous.getChildren().get(childNumber));
+                        break;
+                    }
+                }
+
+                if (count < childNumber) {
+                    throw new FacesException("Component with clientId \""
+                            + previous.getClientId(searchKeywordContext.getSearchExpressionContext().getFacesContext()) 
+                            + "\" has fewer children as \"" + 
+                              childNumber + "\". Expression: \"" + command + "\"");
+                }
+            } else {
+                throw new FacesException(
+                        "Expression does not match following pattern @child(n). Expression: \"" + command + "\"");
+            }
+        } catch (Exception e) {
```

Comment:
Ahh. I can't remember why i added the try/catch around this block as it's actually not required. Will remove it for now.